azure-storage-blob/azure/storage/blob/blob_service_client.py

```diff
@@ -78,7 +88,30 @@ def create_configuration(**kwargs):
 
     def make_url(self, protocol=None, sas_token=None):
         # type: (Optional[str], Optional[str]) -> str
-        pass
+        """
+        Creates the url to access this blob.
+
+        :param str protocol:
+            Protocol to use: 'http' or 'https'. If not specified, uses the
+            protocol specified in the URL when the client was created..
+        :param str sas_token:
+            Shared access signature token created with
+            generate_shared_access_signature.
+        :return: blob access URL.
+        :rtype: str
+        """
+        parsed_url = urlparse(self.url)
+        new_scheme = protocol or parsed_url.scheme
+        query = []
+        if sas_token:
+            query.append(sas_token)
+        new_url = "{}://{}{}".format(
+            new_scheme,
+            parsed_url.netloc,
+            parsed_url.path)
+        if query:
+            new_url += "?{}".format('&'.join(query))
```

Comment:
`query` is only ever empty or just has `sas_token` correct? Unless there's a strong likelihood of this needing generalization later, I'd consider simplifying:
```
if sas_token:
    new_url += "?{}".format(sas_token)
```
and delete the three lines above starting at `query = []`